packages/core/src/sagas/startDictateAfterSpeakActivitySaga.js

```diff
@@ -1,38 +1,43 @@
 import {
-  cancel,
-  fork,
   put,
   select,
-  take
+  takeEvery
 } from 'redux-saga/effects';
 
 import whileConnected from './effects/whileConnected';
+import whileSpeakIncomingActivity from './effects/whileSpeakIncomingActivity';
 
 import { MARK_ACTIVITY } from '../actions/markActivity';
-import { START_SPEAKING_ACTIVITY } from '../actions/startSpeakingActivity';
-import { STOP_SPEAKING_ACTIVITY } from '../actions/stopSpeakingActivity';
 import startDictate from '../actions/startDictate';
 
 export default function* () {
-  yield whileConnected(function* (_, userID) {
-    for (;;) {
-      yield take(START_SPEAKING_ACTIVITY);
-
-      const task = yield fork(startDictateAfterSpeakActivitySaga, userID);
-
-      yield take(STOP_SPEAKING_ACTIVITY);
-      yield cancel(task);
-    }
+  yield whileConnected(function* () {
+    yield whileSpeakIncomingActivity(startDictateAfterSpeakActivitySaga);
   });
 }
 
 function* startDictateAfterSpeakActivitySaga() {
-  for (;;) {
-    const { payload: { activityID } } = yield take(({ payload, type }) => type === MARK_ACTIVITY && payload.name === 'speak' && payload.value === false);
-    const activities = yield select(({ activities }) => activities);
+  yield takeEvery(
+    ({ payload, type }) => (
```

Comment:
inlining this takes away from the readability for me. can it be defined outside the function and given a good name?